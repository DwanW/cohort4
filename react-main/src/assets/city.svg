<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!-- Generator: Adobe Illustrator 19.0.0, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->

<svg
   xmlns:dc="http://purl.org/dc/elements/1.1/"
   xmlns:cc="http://creativecommons.org/ns#"
   xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
   xmlns:svg="http://www.w3.org/2000/svg"
   xmlns="http://www.w3.org/2000/svg"
   xmlns:sodipodi="http://sodipodi.sourceforge.net/DTD/sodipodi-0.dtd"
   xmlns:inkscape="http://www.inkscape.org/namespaces/inkscape"
   version="1.1"
   id="Layer_1"
   x="0px"
   y="0px"
   viewBox="0 0 793.70079 793.7008"
   xml:space="preserve"
   sodipodi:docname="city.svg"
   width="210mm"
   height="210mm"
   inkscape:version="0.92.4 (5da689c313, 2019-01-14)"><metadata
   id="metadata47"><rdf:RDF><cc:Work
       rdf:about=""><dc:format>image/svg+xml</dc:format><dc:type
         rdf:resource="http://purl.org/dc/dcmitype/StillImage" /><dc:title></dc:title></cc:Work></rdf:RDF></metadata><defs
   id="defs45" /><sodipodi:namedview
   pagecolor="#ffffff"
   bordercolor="#666666"
   borderopacity="1"
   objecttolerance="10"
   gridtolerance="10"
   guidetolerance="10"
   inkscape:pageopacity="0"
   inkscape:pageshadow="2"
   inkscape:window-width="1920"
   inkscape:window-height="1051"
   id="namedview43"
   showgrid="false"
   units="mm"
   inkscape:zoom="0.51868132"
   inkscape:cx="150.38136"
   inkscape:cy="458.85593"
   inkscape:window-x="-9"
   inkscape:window-y="-9"
   inkscape:window-maximized="1"
   inkscape:current-layer="Layer_1" />
<g
   id="g10"
   transform="matrix(1.4533898,0,0,1.4491526,73.262715,61.073678)"
   style="fill:#00ffff">
	<polygon
   points="91.25,122.603 0,168.806 0,455 91.25,455 "
   id="polygon2"
   style="fill:#00ffff" />
	<polygon
   points="273.125,102.229 242.5,102.229 242.5,455 333.75,455 333.75,102.229 303.125,102.229 303.125,46.565 273.125,46.565 "
   id="polygon4"
   style="fill:#00ffff" />
	<rect
   x="363.75"
   y="230.94299"
   width="91.25"
   height="224.05701"
   id="rect6"
   style="fill:#00ffff" />
	<rect
   x="121.25"
   width="91.25"
   height="455"
   id="rect8"
   y="0"
   style="fill:#00ffff" />
</g>
<g
   id="g12"
   transform="translate(0,338.70079)">
</g>
<g
   id="g14"
   transform="translate(0,338.70079)">
</g>
<g
   id="g16"
   transform="translate(0,338.70079)">
</g>
<g
   id="g18"
   transform="translate(0,338.70079)">
</g>
<g
   id="g20"
   transform="translate(0,338.70079)">
</g>
<g
   id="g22"
   transform="translate(0,338.70079)">
</g>
<g
   id="g24"
   transform="translate(0,338.70079)">
</g>
<g
   id="g26"
   transform="translate(0,338.70079)">
</g>
<g
   id="g28"
   transform="translate(0,338.70079)">
</g>
<g
   id="g30"
   transform="translate(0,338.70079)">
</g>
<g
   id="g32"
   transform="translate(0,338.70079)">
</g>
<g
   id="g34"
   transform="translate(0,338.70079)">
</g>
<g
   id="g36"
   transform="translate(0,338.70079)">
</g>
<g
   id="g38"
   transform="translate(0,338.70079)">
</g>
<g
   id="g40"
   transform="translate(0,338.70079)">
</g>
</svg>