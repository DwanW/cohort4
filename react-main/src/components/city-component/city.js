import React from 'react';

export const whichSphere = (latitude) => {
    if (Number(latitude) > 0 && Number(latitude) <= 180) {
        return "North Hemisphere";
    } else if (Number(latitude) < 0 && Number(latitude) >= -180) {
        return "South Hemisphere";
    } else if (Number(latitude) === 0) {
        return "Equator";
    } else {
        return "Latitude Input is invalid";
    }
}

export const howBig = (population) => {
    return (population > 100000) ? "City" :
        (population > 20000) ? "Large Town" :
            (population > 1000) ? "Town" :
                (population > 100) ? "Village" :
                    (population > 0) ? "Hamlet" : "Incorrect population input";
}

class City extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            relocate: 0,
        }
        this.handleChange = this.handleChange.bind(this);
        this.movedIn = this.movedIn.bind(this);
        this.movedOut = this.movedOut.bind(this);
    }

    handleChange(e) {
        this.setState({ relocate: e.target.value });
    }
    movedIn() {
        let afterPopulation = Number(this.props.population) + Number(this.state.relocate);
        this.props.onValueChange(afterPopulation, this.props.idx);
    }

    movedOut() {
        let afterPopulation = Number(this.props.population) - Number(this.state.relocate);
        this.props.onValueChange(afterPopulation, this.props.idx);
    }

    render() {
        return (
            <div className="cityCard">
                <div className="cityHeader">{this.props.cityName}</div>
                <input className="inputbox" placeholder="Enter population" type="number" onChange={this.handleChange} />
                <button className="movein city-button" onClick={this.movedIn}>Move In</button>
                <button className="moveout city-button" onClick={this.movedOut}>Move Out</button>
                <div className="result">
                    <div>{`Latitude: ${this.props.cityLat}`}</div>
                    <div>{`Longitude: ${this.props.cityLong}`}</div>
                    <div>{`Population: ${this.props.population}`}</div>
                    <div>{`It is ${howBig(this.props.population)} sized and is located at ${whichSphere(this.props.cityLat)}.`}</div>
                </div>
                <button className="delete city-button" onClick={this.props.onDelete}><i className="fa fa-trash" /></button>
            </div>
        )
    }
};

export default City;