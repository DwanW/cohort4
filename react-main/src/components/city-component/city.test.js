import React from 'react';
import { shallow } from 'enzyme';
import { whichSphere, howBig } from './city';
import City from './city';

describe('test utility functions', () => {
    it('should display correct result for whichSphere function', () => {
        expect(whichSphere(100)).toBe('North Hemisphere');
        expect(whichSphere(-100)).toBe('South Hemisphere');
        expect(whichSphere(0)).toBe('Equator');
        expect(whichSphere(-300)).toBe('Latitude Input is invalid');
    });

    it('should display correct result for howBig function', () => {
        expect(howBig(111111)).toBe("City");
        expect(howBig(25000)).toBe("Large Town");
        expect(howBig(3000)).toBe("Town");
        expect(howBig(500)).toBe("Village");
        expect(howBig(20)).toBe("Hamlet");
        expect(howBig(-30)).toBe("Incorrect population input");
    });
});

describe('City Component testing', () => {
    let wrapper;
    const mockCityName = "Edmonton";
    const mockCityLat = 140;
    const mockCityLong = 40;
    const mockPopulation = 10000;
    let mockOnDelete;
    let mockOnValueChange;
    const mockIdx = 3;
    const mockKey = "56427";

    beforeEach(() => {
        mockOnDelete = jest.fn();
        mockOnValueChange = jest.fn();

        const mockProps = {
            key: mockKey,
            idx: mockIdx,
            cityName: mockCityName,
            cityLat: mockCityLat,
            cityLong: mockCityLong,
            population: mockPopulation,
            onDelete: mockOnDelete,
            onValueChange: mockOnValueChange
        };

        wrapper = shallow(<City {...mockProps} />);
    });

    it('should render city component correctly', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.state()).toEqual({ "relocate": 0 })
    });

    it('should render correct string given the props', () => {
        let resultSection = wrapper.find(".result").children();
        expect(resultSection.length).toBe(4);
        expect(resultSection.at(0).text()).toBe(`Latitude: ${mockCityLat}`);
        expect(resultSection.at(1).text()).toBe(`Longitude: ${mockCityLong}`);
        expect(resultSection.at(2).text()).toBe(`Population: ${mockPopulation}`);
        expect(resultSection.at(3).text()).toBe(`It is ${howBig(mockPopulation)} sized and is located at ${whichSphere(mockCityLat)}.`);
    })

    it('should fire correct action', ()=> {
        wrapper.find('.delete').simulate('click');
        expect(mockOnDelete).toHaveBeenCalled();
        wrapper.find('.movein').simulate('click');
        wrapper.find('.moveout').simulate('click');
        expect(mockOnValueChange).toHaveBeenCalledTimes(2);
    })

})