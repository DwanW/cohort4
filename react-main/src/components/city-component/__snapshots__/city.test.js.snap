// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City Component testing should render city component correctly 1`] = `ShallowWrapper {}`;
